### ISSUE TITLE
Google Drive scope too broad?

### ISSUE THREAD

**Issue by yPhil-gh** [remotestorage/remotestorage.js#1234]
Hi ; I have been using [the remoteStorage widget](https://github.com/remotestorage/remotestorage-widget) in [Pétrolette](https://petrolette.space/) for years and it works great (quite literally) and now I want to offer Google Drive sync and it apparently works too :) but despite having reduced the scope to `/auth/drive.file` (API User-facing description: "See, edit, create, and delete only the specific Google Drive files you use with this app") the Auth page still asks for full access, wich is both scary and not true, since my app only needs to create _one_ dir and read / write to _one_ file in it.

I noticed that the scope is [hard-coded](https://github.com/remotestorage/remotestorage.js/blob/dd1b9f5848b3c60435837ec7c0d48d8933cdc403/src/googledrive.ts#L16) in https://github.com/remotestorage/remotestorage.js/blob/master/src/googledrive.ts ; Is there any way to override it?

**Comment by raucao** [comment]
Is that a new API feature? When we added Google Drive support, there was no scope available for a single folder. It would actually be amazing if we could add that now, in case Google finally allow it, after almost a decade of making this impossible for their users!

**Comment by yPhil-gh** [comment]
> Is that a new API feature? When we added Google Drive support, there was no scope available for a single, arbitrary folder.

I think it is, because I seem to remember the API scope list being shorter when I checked it out, like at least 5 years ago.

> It would actually be amazing if we could add that now, in case Google finally allow it, after almost a decade of making this impossible for their users!

It really would! If I may offer help, I could update [the relevant part of the documentation](https://remotestoragejs.readthedocs.io/en/latest/getting-started/dropbox-and-google-drive.html) to reflect the changes.

**Comment by raucao** [comment]
Yes, any help would be most appreciated. Shouldn't be hard to change the Google Drive backend to use the scope defined in `remoteStorage.access` either.

We're currently in the process of finalizing the 2.0.0 beta. So this could still go in the 2.0 release I think.

**Comment by yPhil-gh** [comment]
> Could you link the relevant Google API docs perhaps?

I sure could: https://developers.google.com/identity/protocols/oauth2/scopes

**Comment by raucao** [comment]
Hmm, I cannot see how we could actually define the folder to access in the relevant documentation:

https://developers.google.com/drive/api/v3/about-auth#OAuth2Authorizing

It sounds to me like a user would have to manually pick a single file during the auth dialog on the Google side or something.

**Comment by yPhil-gh** [comment]
> It sounds to me like a user would have to manually pick a single file during the auth dialog on the Google side or something.

To be honest I'm already flabbergasted that it works so transparently for 5apps, and I wouldn't mind explaining the procedure you mentioned to my users ; After all it's [the spirit of the documentation](https://remotestoragejs.readthedocs.io/en/latest/getting-started/dropbox-and-google-drive.html) :

> However, while RS is not a widely known and deployed protocol, we find it helpful to let users choose something they already know, and potentially migrate to an RS account later on.

**Comment by raucao** [comment]
It might still be an option, but it's certainly neither as straight-forward to implement in the library, nor to use for end users, as a predefined directory access scope would be. We'll have to look into it further. Ideally someone could create a PR with a proof-of-concept to test with different apps and scenarios.
